### DISCUSSION TITLE
[Question]: Does this mean that with nixci we can use things like sub-flake and monorepo in our nix projects?

### DISCUSSION THREAD

**Discussion by I-Want-ToBelieve** [srid/nixci#56] [Q&A] [answered]
If so, that's really awesome, and after the advent of nix flake, supporting sub-flake and monorepo was pretty much a must-go

It would be even better if it also supported task orchestration and compilation caching

**Comment by srid** [comment]
Yes, have you [checked the examples](https://github.com/srid/nixci?tab=readme-ov-file#examples)? They do use sub-flakes. 

> It would be even better if it also supported task orchestration and compilation caching

Could you describe in detail what you mean?

**Reply by I-Want-ToBelieve** [reply]
compilation caching:
Only build sub-flakes that have changed content

task orchestration:
If there are no dependencies between sub-flakes, then build all sub-flakes as concurrently as possible, if there are dependencies, then orchestrate tasks, build sequentially for sub-flakes with dependencies, and build concurrently as much as possible for those without dependencies.

**Reply by srid** [reply, marked as answer]
> Only build sub-flakes that have changed content

Doesn't it already do this?

> task orchestration:

`-j <num>`?
